build: align VSCode extension build with vsce

alan-agius4 opened the pull request:
This commit revamps the debugging setup and enabling developers to set breakpoints directly in the source TypeScript files.

Key changes include:
- Updated `launch.json` with source map path overrides to correctly map compiled output back to the original source code.
- Switched from `external` to `linked` sourcemaps in the Bazel build configuration for better debugging support.
- Consolidated the recommended VSCode settings into the main `launch.json` and `tasks.json`, removing the separate `recommended-*.json` files.
- Updated the debugging documentation to reflect the new, simplified workflow.

These changes significantly improve the developer experience for contributors working on the language service, making it much easier to debug and troubleshoot issues.

This applies to both the framework packages and vscode-ng-langugage-service.

alan-agius4 committed:
build: align VSCode extension build with vsce

This commit refactors the VSCode extension's build and launch configurations to align with `vsce` and `pnpm` workspaces.

The following changes are included:
- Updated `.vscode/launch.json` to use new `vsce:` prefixed tasks and simplified launch configurations.
- Updated `.vscode/tasks.json` to use `pnpm` workspace commands for watching and packaging the extension.
- Adjusted `BUILD.bazel` and `package.json` files to reflect the new build output paths.

These changes streamline the development workflow for the VSCode extension, making it easier to build, debug, and package.

alan-agius4 committed:
build: update supported Node.js versions for language-service

The supported Node.js versions for the language service server are updated to align with the versions supported by Angular. This ensures a consistent and stable environment for developers.

atscott commented on .vscode/launch.json:
```diff
@@ -5,90 +5,44 @@
       ],
       "outFiles": [
-        "${workspaceFolder}/dist/bin/vscode-ng-language-service/client/src/*.js"
+        "${workspaceFolder}/dist/bin/vscode-ng-language-service/client/index.js",
+        "${workspaceFolder}/dist/bin/vscode-ng-language-service/server/index.js"
```
Is this right? I think ideally it includes all the files, not just the final, single index.js, which I would think is more appropriate for the "launch prod client" task

alan-agius4 replied:
The outFiles are for break points to work. the `index.js` is the bundle containing all the contaminated files transpired TS files, the map of the index.js points to the TS files and not the transitive JS files generated by `ts_project`.

atscott replied:
Hmm, okay. When we switched to bazel in the vscode repo, we had to change our breakpoints to be the transitive js files in the bazel-bin. IIUC, what you're saying will change this again (totally fine, just trying to understand)?

atscott replied:
I just spent some time debugging an issue and think it would be much easier if we could avoid using the bundled version. I wasn't able to get breakpoints working in the TS file so I had to dig through the index.js to find where to put breakpoints (much harder, especially since it includes the bundled TS library)

This isn't a dealbreaker at the moment - what you've done to fix this so far is great on its own!

alan-agius4 replied:
I actually spent sometime looking at this and the problem is that the slice maps are pointing to the TS files located into the bazel-out and not the src. I have some ideas on how to solve this both with bundling and without. (Fingers crossed).

I would be more inclined towards the bundling approach; as it closer to production, example it uses the `banner.ts` as the dependencies are resolved and inlined like the proper bundle vs using node.js resolution and using the CJS and AMD formats.

I’ll continue on this on Monday as it’s getting pretty late for me.

alan-agius4 replied:
I think I have an idea of how to make thie break out in the src work, let me try it on Monday, I’ll keep you posted.

alan-agius4 commented:
@atscott, I have made some changes and now breakpoints work from source, this also now works for the framework tests and packages.

alan-agius4 committed:
build: enable breakpoints in source files

This commit revamps the debugging setup and enabling developers to set breakpoints directly in the source TypeScript files.

Key changes include:
- Updated `launch.json` with source map path overrides to correctly map compiled output back to the original source code.
- Switched from `external` to `linked` sourcemaps in the Bazel build configuration for better debugging support.
- Consolidated the recommended VSCode settings into the main `launch.json` and `tasks.json`, removing the separate `recommended-*.json` files.
- Updated the debugging documentation to reflect the new, simplified workflow.

These changes significantly improve the developer experience for contributors working on the language service, making it much easier to debug and troubleshoot issues.

This applies to both the framework packages and vscode-ng-langugage-service.

alan-agius4 committed:
build: update repository URL to point to monorepo

The repository URL in `package.json` is updated to point to the `angular/angular` monorepo. This ensures that the package metadata correctly reflects its location within the main repository.

atscott commented on .gitignore:
```diff
@@ -16,7 +16,6 @@ pubspec.lock
 !.devcontainer/recommended-devcontainer.json
 !.devcontainer/recommended-Dockerfile
 .settings/
-.vscode/settings.json
```
this one might be controversial. @JeanMeche added it recently

JeanMeche replied:
Yeah there was some little back-and-forth on this. Initially removed by @josephperrott in #63924, I added it back because I believed it wasn't necessary for the extension + `.vscode/settings.json` contains some local configurations (in my case it's the Vscode colors, to distinguish the projects I'm working on).

alan-agius4 committed:
fixup! build: enable breakpoints in source files

mmalerba commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: dd7bb9007f0e222602c38953564a61610a9791df
